src/HypLSP7.sol

```diff
@@ -57,9 +58,9 @@ contract HypLSP7 is LSP7DigitalAssetInitAbstract, TokenRouter {
             isNonDivisible_: false // isNonDivisible set to `false` as not used anyway since decimals() is overriden
          });
 
-        // emit `DataChanged` event only if some metadata bytes is provided to save gas
-        if (_lsp4Metadata.length > 0) {
-            _setData(_LSP4_METADATA_KEY, _lsp4Metadata);
+        // set init data keys & values
+        if (dataKeys.length > 0 && dataKeys.length == dataValues.length) {
```

Comment:
Hmmm is this second check needed?
If I provide 2 x data keys and 3 x data values for instance, this check from `ERC725Y` will be skipped.

https://github.com/ERC725Alliance/ERC725/blob/5b9ff11678bcaba4ce31d4699b9f9fa484b3251d/implementations/contracts/ERC725Y.sol#L181-L194

<img width="1012" alt="image" src="https://github.com/user-attachments/assets/29943f5f-831c-408e-a86f-d496b3d2c45a" />

Reply:
Suggest adding a test for this case above. That it should revert in case